triqs/gfs/block/block_gf.hxx

```diff
@@ -209,6 +212,21 @@ namespace triqs {
       /// Construct from the vector of names
       block_gf(block_names_t b) : _block_names(std::move(b)), _glist(_block_names.size()) {}
 
+      // Create Block Green function from Mesh and gf_struct
+      block_gf(gf_mesh<Var> const &m, triqs::hilbert_space::gf_struct_t const &gf_struct) {
+
+        for (auto const & [ bname, idx_lst ] : gf_struct) {
+          auto bl_size = idx_lst.size();
+          _block_names.push_back(bname);
+          std::vector<std::string> idx_str_lst(bl_size);
+          std::transform(idx_lst.cbegin(), idx_lst.cend(), idx_str_lst.begin(), [](auto &arg) { return std::to_string(arg); });
```

Comment:
I am not sure which version is really better here. I find the above syntax quite clear. Which advantages do you see with the back inserter?

Reply:
> I am not sure which version is really better here.

Me neither.
Okay, my idea was to avoid allocations of empty strings in `std::vector<std::string> idx_str_lst(bl_size)`. But I guess that's unimportant after all.